From 905273eb05e62da59c0f3c7f07fe605910afaef9 Mon Sep 17 00:00:00 2001
From: Simon Dubray <simonx.dubray@intel.com>
Date: Wed, 22 Apr 2015 17:44:59 +0200
Subject: [PATCH] Fix kw issues in PM code

KW fixed :
- arch/x86/kernel/intel_nc_dev_state.c:129 nc_dev_state_list()
  Pointer 's' checked for NULL at line 116 may be dereferenced at line 129
- arch/x86/kernel/pmc_atom.c:350 pmc_dev_state()
  Pointer 's' checked for NULL at line 320 may be dereferenced at line 350

Change-Id: I5e63ad550820cde2ba68ea727d41470f26622aa1
Signed-off-by: Simon Dubray <simonx.dubray@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-33477
---
 arch/x86/kernel/intel_nc_dev_state.c | 2 +-
 arch/x86/kernel/pmc_atom.c           | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kernel/intel_nc_dev_state.c b/arch/x86/kernel/intel_nc_dev_state.c
index 8dc110f55a34..45275cdb27a5 100644
--- a/arch/x86/kernel/intel_nc_dev_state.c
+++ b/arch/x86/kernel/intel_nc_dev_state.c
@@ -125,7 +125,7 @@ static void nc_dev_state_list(void *seq_file)
 			if (!nc_val)
 				pr_info("%s in NC is in D0 prior to suspend\n",
 				nc_dev[dev_index].name);
-		} else
+		} else if (s)
 			seq_printf(s, "%9s : %s\n", nc_dev[dev_index].name,
 				dstates[nc_val]);
 	}
diff --git a/arch/x86/kernel/pmc_atom.c b/arch/x86/kernel/pmc_atom.c
index 64d8df792b17..946764b4fbbc 100644
--- a/arch/x86/kernel/pmc_atom.c
+++ b/arch/x86/kernel/pmc_atom.c
@@ -346,7 +346,7 @@ static void pmc_dev_state(void *seq_file)
 				pr_info("%s in SC is in D0 prior to sleep\n",
 					dev_map[dev_index].name);
 
-		} else {
+		} else if (s) {
 			seq_printf(s, "Dev: %-32s\tState: %s [%s]\n",
 			dev_map[dev_index].name,
 			func_dis_index ?
-- 
1.9.1

